pkg/services/lambda.go

```diff
@@ -0,0 +1,46 @@
+package services
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/lambda"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+)
+
+type generateRequest struct {
+	HTML string `json:"html"`
+}
+
+type generateResponse struct {
+	Content []byte `json:"content"`
+}
+
+// NewInvokeGeneratePDF returns a function that saves the metadata of an uploaded file
+func NewInvokeGeneratePDF(config Config, client *lambda.Lambda) func(cxt context.Context, html string) ([]byte, error) {
+	return func(ctx context.Context, html string) ([]byte, error) {
+		appcontext.ZLogger(ctx).Info("making request to lambda")
+
+		var request generateRequest
+		payload, marshalErr := json.Marshal(request)
+		if marshalErr != nil {
+			return nil, fmt.Errorf("error marshaling generateRequest: %w", marshalErr)
+		}
+
+		result, invokeErr := client.Invoke(&lambda.InvokeInput{FunctionName: aws.String("main"), Payload: payload})
```

Comment:
How does the function name get passed in?

Reply:
Good question. I think this is where the function name is passed in, but since this code was written against `docker-lambda` that was the value that was needed then. I imagine we need another config value for this, though, since the name will be different in other environments.

I'm going to get this working against the hosted dev lambda and will push a code update with any changes.

Reply:
The latest code looks at the `LAMBDA_FUNCTION_PRINCE` env var for the name of the function to invoke.